### ISSUE TITLE
Bylvis responds with after-quest text before quest completed

### ISSUE THREAD

**Issue by mozai** [Wynncraft/Issues#2544]
Problem: Quest NPC Bylvis will emit [the post-quest message] "You did give him the emeralds... Right?" before the associated quest is completed.

Steps to reproduce:

- New player character, any class.  Advance to level 15.
- Speak to Bylvis at Katoa Ranch.  page through the thirteen messages that start the quest, last message is "[13/13] Bylvis: Hey you! Can you ..."
- Speak to Bylvis again **and see incorrect reply**.
- Walk to the cattle stall for Drale, trigger the event prompting you to talk to Bylvis.
- Speak to Bylvis again, page through to "[5/5] Really? Thanks a lot. Now..."
- Speak to Bylvis again **and see incorrect reply**.

Screenshot:

![Screenshot](http://i.imgur.com/MPc1BE1.png)

**Comment by Moe-Ronickah** [comment]
Mozai, the Confused Farmer quest needs an update, however, I have no idea if it will get one.  You would expect Blyvis to tell you to go see Drale after telling you to go see him, however, he doesn't.  Instead, the quest assumes that you went to see Drale and got your quest rewards from him.  Instead, it looks as if the quest was written, considered to be long enough, and then ended at Drale.  In order for Blyvis to say "Go see Drale!", Drale would have to send you back to Blyvis in order to complete the quest.  So, actually, the quest is working as it was written.  If I've helped to explain this mess, and it is a mess, please close this issue.  Thanks.

**Comment by mozai** [comment]
The quest doesn't go Blyvis - proximity - Blyvis - Drale - end.  The quest steps are:

 -(before quest) Blyvis has Message #B0 (1 line)
 -(before quest) Drale has Message #D0 (1 line)
 -a Talk to Blyvis, get Message #B1 (13 lines)
 -b Proximity trigger (pressure plate?) in Katoa.
 -c Talk to Blyvis, get Message #B2 (5 lines)
 -d Talk to Drale, get Message #D1 (9 lines)
 -e Talk to Blyvis, get Mesage #B3 (6 lines)
 -f Blyvis has message #B4 (1 line)
 -g Talk to Drale, get Message #D2 (7 lines)
 -h Drale has message #D3 (1 line)

The bug is Blyvis is emitting message #B4 immediately after step a, before step f, before the quest is over and when there are still state changes pending in the quest.

If "as written" means "intentional," I disagree that it makes sense for Blyvis to ask you about emeralds that he hasn't given to you yet.  If "as written" means "as close as we could get to what we intended, but we're limited by the tools we have," then you can mark this bug as WONTFIX, or link this issue to the "improve the code for handling quests" issue if there is one.

**Comment by Moe-Ronickah** [comment]
I don't mark quests as anything, I'm not an admin.  I just help where I can, and confuse things, where I can't.  As I said, "You would expect Blyvis to tell you to go see Drale after telling you to go see him, however, he doesn't."  Once you talk to Bylvis for the third time, the only thing that Bylvis says is "You did give him . . .?"  Because the quest ends with Dale giving you the major quest rewards, XP and some more emeralds, the only thing that Bylvis has to say to you is "You did give him . . . ?"  It's a low level quest.  If the player was sent by Dale BACK to Bylvis to complete the quest, then Bylvis could say something else, BEFORE we talk to Drale, like "Go see Drale, OK?" or "You need to go speak with Drale", and apparently that extra travel was deemed to be too much for a low level quest at the time that the quest was made or the quest was finished in a hurry and left as it is.

Will it get changed?  I have no idea.

**Comment by DarkDemonDragon** [comment]
This bug also drives home the fact that you DON'T give Drale anything as this dialogue happens:
Drale: So I just spent three years of my life imprisoned for nothing. Unless of course, he sent me something as an apology?
Drale: Nothing? Really? Not even a few emeralds?

I just try to ignore this whole thing as what do you expect when the quest is called "The Confused Farmer" XD So many people don't finish this quest because they never get that Drale isn't a cow >_<

**Comment by mozai** [comment]
Moe, you keep mis-reading the problem description.  You said "once you talk to Blyvis for the third time..." but this message is happening before the third time (step e, above), and before the scripted second time (step c, above).  *That's the bug.*   
Another description: Other quest NPCs can have at least three responses other than quest instructions (ie. prequest "come back when you're level xx", duringquest "have you found the macguffin yet?", postquest "thank you for the help") but Blyvis seems only to have two responses.

**Comment by DarkDemonDragon** [comment]
I feel like this quest reveals some hidden insight on who the "player" is. Whenever you receive emeralds from a quest, a Quest NPCs almost always says something like "Here's your reward" or "Take these and go". However, in this quest you receive emeralds without anyone saying anything conclusive. The only 2 NPCs in this quest are Blyvis and Drale. 

Now we know that Drale probably didn't give you emeralds because he's demanding them, plus while you're doing him a slight favor by clearing things up he really has no reason to reward you. 

With Blyvis however it's different. While the quest ends at Drale, the only NPC that really could have given you any emeralds is Blyvis when he asks you to give Drale some emeralds. You "hold" onto them until you talk to Drale, but you don't actually "receive" and keep the emeralds until you finish up at Drale. Drale even says:
Drale: So I just spent three years of my life imprisoned for nothing. Unless of course, he sent me something as an apology?
Drale: Nothing? Really? Not even a few emeralds?

Obviously there is a contradiction here because Blyvis specifically asks you to deliver emeralds as an apology. But Drale doesn't know this. This means that the "Player" took advantage of the situation and, knowing that neither parties would ever contact each other about this, kept the emeralds for himself. 

So in conclusion either the Player is a d*ck or this is a bug.

**Comment by Moe-Ronickah** [comment]
mozai, I apologize for not correctly reading what you posted and not paying clear attention to your screen captures.  Yes, there is definitely a bug in the conversation script between Bylvis and the player that needs to be addressed.  Bylvis should not be saying "You did give him the emeralds... Right?" until after the third time that the player interacts with Bylvis.  Thank you for pointing this out.

DarkDemonDragon, your second comment would be better addressed in the forums, in the "Suggestions" section.

**Comment by DarkDemonDragon** [comment]
@Moe-Ronickah ehh it's not really a suggestion though since I'm just interpreting what's happening. Who knows? Maybe this was slightly intentional or the staff realized this bug but decided it actually added to the lore of Wynncraft. I prefer not to make any public theory posts since my highest lvl class is lvl 52 so I haven't done all the quest nor have I carefully read through the existing official lore. (I've only skimmed there was a guy named Bob and he was awesome. He died. Feel sad about it)

**Comment by Trixter313** [comment]
Are you still experiencing this since the Gavel update?

If your problem is now resolved, we'd appreciate it if you closed this issue. If it's not, we'd be happy to continue to assist you as much as possible. Thank you and we hope you enjoy your time playing Wynncraft! :)

**Comment by DarkDemonDragon** [comment]
@Trixter313 is this a bug or intentional? Also, this is annoying to retest as one has to make a new character and remember to screenshot everything once they can do this quest :/

**Comment by Moe-Ronickah** [comment]
I've started a new character, so I can test this when I get to it.  Let me see what I get, and I'll let everyone know what the result is, provided that I can remember to check this.

**Comment by Moe-Ronickah** [comment]
I am posting the pictures, here, from completing this quest in the past 12 hours.  Bylvis's lines still need to be fixed.
![2015-12-24_00 03 43](https://cloud.githubusercontent.com/assets/11842380/11996631/897660b4-aa2c-11e5-9187-007daa4a72f6.png)
![2015-12-24_00 04 44](https://cloud.githubusercontent.com/assets/11842380/11996632/9413f3b0-aa2c-11e5-91a4-e9c0f2f912ee.png)
![2015-12-24_00 10 24](https://cloud.githubusercontent.com/assets/11842380/11996633/9a34b022-aa2c-11e5-98d5-642d57a03196.png)
![2015-12-24_00 10 38](https://cloud.githubusercontent.com/assets/11842380/11996634/9d733c86-aa2c-11e5-8c0f-298dceaaafbf.png)
![2015-12-24_00 14 57](https://cloud.githubusercontent.com/assets/11842380/11996635/a30d783c-aa2c-11e5-9d21-8f6566557cf2.png)
![2015-12-24_00 18 39](https://cloud.githubusercontent.com/assets/11842380/11996640/b2fc3d82-aa2c-11e5-8254-68c01e21bf50.png)

**Comment by DarkDemonDragon** [comment]
@Tama63 I realize that you've done a mass close because the issues were piling up, but please revisit this as this issue still persists. Post-Gavel evidence has been provided recently too, so please look into this.
